dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/task.py

```diff
@@ -175,18 +176,28 @@ def process_definition(self, process_definition: Optional[ProcessDefinition]):
     def resource_list(self) -> List:
         """Get task define attribute `resource_list`."""
         resources = set()
-        for resource in self._resource_list:
-            if type(resource) == str:
-                resources.add(self.query_resource(resource).get(ResourceKey.ID))
-            elif type(resource) == dict and resource.get(ResourceKey.ID) is not None:
+        for res in self._resource_list:
+            if type(res) == str:
+                resources.add(
+                    Resource(name=res, user_name=self.user_name).get_id_from_database()
+                )
+            elif type(res) == dict and res.get(ResourceKey.ID) is not None:
                 logger.warning(
                     """`resource_list` should be defined using List[str] with resource paths,
                        the use of ids to define resources will be remove in version 3.2.0.
                     """
                 )
-                resources.add(resource.get(ResourceKey.ID))
+                resources.add(res.get(ResourceKey.ID))
         return [{ResourceKey.ID: r} for r in resources]
 
+    @property
+    def user_name(self) -> Optional[str]:
+        """Return user name of process definition."""
+        if self.process_definition:
+            return self.process_definition.user.name
+        else:
+            return None
```

Comment:
I think we can directly raise error here, because we will still raise error in function `get_info_from_database` when `user_name` is none